### DISCUSSION TITLE
Client IP variable for runOnRead and HLS/RTMP support?

### DISCUSSION THREAD

**Discussion by gluedots** [bluenviron/mediamtx#570] [Questions]
Hello, would it be possible to expose a readIP variable to the runOnRead command?

I also noticed that runOnRead doesn't seem to trigger when accessing the HLS or RTMP streams. It works great for RTSP, but can't figure out how to run a command when reading HLS or RTMP. Could this be added too?

Thanks!

**Comment by aler9** [comment]
> readIP variable to the runOnRead command

open a feature request for this.

> I also noticed that runOnRead doesn't seem to trigger when accessing the HLS or RTMP streams. It works great for RTSP, but can't figure out how to run a command when reading HLS or RTMP. Could this be added too?

Thanks for reporting the issue, now `runOnRead` is run also when someone reads a stream with RTMP. Unfortunately it can't be run in case of HLS clients, since in HLS there isn't a concept of client, the video is just split into segments and made available through an HTTP server.

[a comment by an automated account is not shown]
